### PR TITLE
[Add] Versions variation and beta-releases support

### PR DESCRIPTION
**Pull request by rudra-iitm**
- The "snapmodule.py" script has been revised to incorporate a new format token, namely '%V' and its prefixed version "prefix/%V."

- This token is applicable in cases where tag versions deviate from the standard or exhibit uncommon characteristics, such as containing alphanumeric values or exceeding three integers.

- When the tag version adheres to the format "1.0b2," the recommended token is specified as `format: "%V"`

- In the scenario where the tag version is structured as "debian/3.22.10+dfsg0-4," it is advised to employ the `format: "debian/%V" `token.

### CODE REVIEW

File: updatesnap/SnapModule/snapmodule.py [hunk 1]
```diff
@@ -667,7 +677,7 @@ def process_part(self, part: str) -> Optional[dict]:
 
         if ("format" not in version_format) and (current_tag is not None):
             # if the version format is not specified,
-            # automagically detect it between any of these common formats:
```
Comment (sergio-costas) [inline updatesnap/SnapModule/snapmodule.py]: It's not a typo, but a conscious choice ;-)

File: updatesnap/SnapModule/snapmodule.py [hunk 2]
```diff
@@ -115,6 +115,13 @@ def _get_version(self, part_name, entry, entry_format, check):
                                   "Missing tag version format for "
                                   f"{part_name}.")
             return None  # unknown format
+        # Check for variations in the version format and beta releases.
+        if '%V' in entry_format['format']:
+            version = pkg_resources.parse_version(entry.split("/")[-1])
```
Comment (sergio-costas) [inline updatesnap/SnapModule/snapmodule.py]: I don't see how this manages to only parse those entries that begin with the specified format. For example, if we specify that format is "debian%V", this code will return any entry, not only those that begin with "debian".

File: updatesnap/SnapModule/snapmodule.py [hunk 3]
```diff
@@ -394,6 +401,9 @@ def _is_gitlab(self, repository):
         """ Evaluates the URI of a repository and returns an URI
             object with it, but only if it is a Gitlab URI. """
         uri = self._get_uri(repository, 3)
+        # Check for gitlab instance used by debian
```
Comment (sergio-costas) [inline updatesnap/SnapModule/snapmodule.py]: Good point, but maybe this should be in a separate MR.

**Review by sergio-costas — CHANGES_REQUESTED** [review]
The idea is good, but I think that some changes are required.

**Comment by rudra-iitm** [comment]
I have discerned the cause of the check failures; it pertains to the obsolete tag list within the `'get_gnome_calculator_tags()' `function in /updatesnap/unittests.py.

File: updatesnap/SnapModule/snapmodule.py (hunk 3 above)
Comment (rudra-iitm) [reply updatesnap/SnapModule/snapmodule.py]: Ok, I will create a separate PR for this

**Review by sergio-costas — APPROVED** [review]
This looks better.

**Comment by sergio-costas** [comment]
Can you fix the test, please?

**Comment by rudra-iitm** [comment]
> Can you fix the test, please?

Already raised a PR #490

**Comment by rudra-iitm** [comment]
@sergio-costas , would you mind re-running the checks, please?

**Comment by sergio-costas** [comment]
@Rudra-IITM Please, rebase this to add the fixes in unitary tests.

File: candidate-snaps-review/candidate.yml [hunk 4]
```diff
@@ -4,7 +4,8 @@ chromium: []
 eog: []
 epiphany: []
 evince: []
-firefox: []
+firefox:
```
Comment (sergio-costas) [inline candidate-snaps-review/candidate.yml:7]: Please, don't mix patches. For this time I'll pass it, but it's not a good idea.

**Comment by rudra-iitm** [comment]
@sergio-costas, could you kindly merge this pull request? All the checks have successfully passed
